plugins/modules/s3_bucket.py

```diff
@@ -1085,6 +1129,7 @@ def main():
         acl=dict(type='str', choices=['private', 'public-read', 'public-read-write', 'authenticated-read']),
         validate_bucket_name=dict(type='bool', default=True),
         dualstack=dict(default=False, type="bool"),
+        object_lock_enabled=dict(type="bool"),
```

Comment:
Dropped the default, if the option's not passed we don't want it reverting things